pkg/controller/nodeipam/node_ipam_controller.go

```diff
@@ -141,7 +141,7 @@ func (nc *Controller) Run(ctx context.Context) {
 	klog.FromContext(ctx).Info("Starting ipam controller")
 	defer klog.FromContext(ctx).Info("Shutting down ipam controller")
 
-	if !cache.WaitForNamedCacheSync("node", ctx.Done(), nc.nodeInformerSynced) {
+	if !cache.WaitForNamedCacheSyncWithContext(ctx, nc.nodeInformerSynced) {
```

Comment:
Can we please do the cloud controllers in a separate PR? And ask somebody from cloud provider to double check this.

For context, the controller name will not be passed via ctx when used in CCM (cloud-controller-manager). Please see https://github.com/kubernetes/kubernetes/blob/f42b497cf25548aa0f327c675e11c57240bfab4b/cmd/cloud-controller-manager/main.go#L46-L80

CCM AFAIU is basically deprecated and has never been updated to use ControllerDescriptors and the controller initialization/start logic that is in KCM.

Reply:
Thank you for explaining the legacy CCM startup logic.
I've updated this PR to remove the changes related to the nodelifecycle and nodeipam controllers as you suggested. This PR is now focused only on the generic, non-cloud controllers.

I will address the cloud-related controllers in a separate PR.
Ready for another look!